dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -25,8 +36,132 @@ namespace ErrorCodes
 {
 extern const int BAD_ARGUMENTS;
 extern const int NOT_IMPLEMENTED;
+extern const int CANNOT_INSERT_NULL_IN_ORDINARY_COLUMN;
 } // namespace ErrorCodes
 
+namespace
+{
+template <typename T>
+consteval bool checkIfSimpleNumericType()
+{
+    return std::is_integral_v<T> || std::is_floating_point_v<T>;
+}
+
+template <typename T>
+struct DecimalWithScale
+{
+    DecimalWithScale(T decimal_, Int64 scale_)
+        : decimal(decimal_)
+        , scale(scale_)
+    {
+        if constexpr (!IsDecimal<T>)
+            throw Exception("DecimalWithScale only accepts Decimal type.");
+    }
+
+    T decimal; // This T should be decimal type
+    Int64 scale;
+};
+
+using DecimalWithScale32 = DecimalWithScale<Decimal32>;
+using DecimalWithScale64 = DecimalWithScale<Decimal64>;
+using DecimalWithScale128 = DecimalWithScale<Decimal128>;
+using DecimalWithScale256 = DecimalWithScale<Decimal256>;
+
+template <typename T>
+consteval bool checkIfDecimalWithScaleType()
+{
+    return std::is_same_v<T, DecimalWithScale32> || std::is_same_v<T, DecimalWithScale64> || std::is_same_v<T, DecimalWithScale128> || std::is_same_v<T, DecimalWithScale256>;
+}
+
+// When T is Decimal, we should convert it to DecimalWithScale type
+// as we need scale value when executing the comparison operation.
+template <typename T>
+struct ActualCmpDataType
+{
+    using Type = std::conditional_t<checkIfSimpleNumericType<T>(), T, DecimalWithScale<T>>;
+};
+
+// ColumnConst should not appear at here as MySQL does not allow the order by column is const, so we do not handle it.
```

Comment:
![xsMloCmevd](https://github.com/pingcap/tiflash/assets/24507159/78b11c75-76e5-4a23-b962-95c761952cf8)
In this plan, if Projection_11 is on ExchangeReceiver_14, order by column const.
The fact that Projection_11 is on ExchangeReceiver_14 does not affect the semantics of the query.

Reply:
![OgS3VJLnOY](https://github.com/pingcap/tiflash/assets/24507159/d31d64a5-65fd-49c3-a9b4-a412002fcc86)
Here comes `order by ColumnConst`, see Sort_21

Reply:
Now, we get value with Field.